tests/test_simulation.py

```diff
@@ -0,0 +1,323 @@
+        assert type(event_1.module) is type(event_2.module)  # noqa: E721
+
+
+def _check_hsi_events_are_equal(hsi_event_1: HSI_Event, hsi_event_2: HSI_Event) -> None:
+    if isinstance(hsi_event_1.target, Population):
+        # We don't check for equality of populations here as we do separately and
+        # it would create a lot of redundancy to check for every HSI event
+        assert isinstance(hsi_event_2.target, Population)
+    else:
+        assert hsi_event_1.target == hsi_event_2.target
+    assert hsi_event_1.module.name == hsi_event_2.module.name
+    assert hsi_event_1.TREATMENT_ID == hsi_event_2.TREATMENT_ID
+    assert hsi_event_1.ACCEPTED_FACILITY_LEVEL == hsi_event_2.ACCEPTED_FACILITY_LEVEL
+    assert hsi_event_1.BEDDAYS_FOOTPRINT == hsi_event_2.BEDDAYS_FOOTPRINT
+    assert (
+        hsi_event_1._received_info_about_bed_days
+        == hsi_event_2._received_info_about_bed_days
+    )
+    assert hsi_event_1.expected_time_requests == hsi_event_2.expected_time_requests
+    assert hsi_event_1.facility_info == hsi_event_2.facility_info
+
+
+def _check_hsi_event_queues_are_equal(
+    hsi_event_queue_1: List[HSIEventQueueItem],
+    hsi_event_queue_2: List[HSIEventQueueItem],
+) -> None:
+    assert len(hsi_event_queue_1) == len(hsi_event_queue_2)
+    for hsi_event_queue_item_1, hsi_event_queue_item_2 in zip(
+        hsi_event_queue_1, hsi_event_queue_2
+    ):
+        assert hsi_event_queue_item_1.priority == hsi_event_queue_item_2.priority
+        assert hsi_event_queue_item_1.topen == hsi_event_queue_item_2.topen
+        assert (
+            hsi_event_queue_item_1.rand_queue_counter
+            == hsi_event_queue_item_2.rand_queue_counter
+        )
+        assert hsi_event_queue_item_1.tclose == hsi_event_queue_item_2.tclose
+        _check_hsi_events_are_equal(
+            hsi_event_queue_item_1.hsi_event, hsi_event_queue_item_2.hsi_event
+        )
+
+
+def _check_simulations_are_equal(
+    simulation_1: Simulation, simulation_2: Simulation
+) -> None:
+    _check_basic_simulation_attributes_equal(simulation_1, simulation_2)
+    _check_modules_are_equal(simulation_1.modules, simulation_2.modules)
+    _check_random_state_equal(simulation_1.rng, simulation_2.rng)
+    _check_event_queues_are_equal(simulation_1.event_queue, simulation_2.event_queue)
+    _check_hsi_event_queues_are_equal(
+        simulation_1.modules["HealthSystem"].HSI_EVENT_QUEUE,
+        simulation_2.modules["HealthSystem"].HSI_EVENT_QUEUE,
+    )
+    _check_population_equal(simulation_1.population, simulation_2.population)
+
+
+@pytest.fixture(scope="module")
+def resource_file_path():
+    return Path(__file__).parents[1] / "resources"
+
+
+@pytest.fixture(scope="module")
+def initial_population_size():
+    return 1000
+
+
+@pytest.fixture(scope="module")
+def start_date():
+    return Date(2010, 1, 1)
+
+
+@pytest.fixture(scope="module")
+def end_date(start_date):
+    return start_date + DateOffset(days=60)
```

Comment:
These tests are great. How about running a slightly longer & larger sim? (or only when running tests on GH...don't know if there's a good way to detect that)

Reply:
Have increased initial population size to 5000 and simulation length to 180 days - with these changes tests in this module still run in 8 minutes distributed across 4 processes locally (presumably will be more like ~40 minutes on single process on Actions runner).

Reply:
It actually only took 11 minutes still, which I guess is partly because of shared simulation fixtures meaning there was limited speed up from running across multiple processes here. We could therefore think about increasing further, though for now I'll leave.